libraries/AP_Compass/AP_Compass_UAVCAN.cpp

```diff
@@ -144,6 +153,21 @@ AP_Compass_UAVCAN* AP_Compass_UAVCAN::get_uavcan_backend(AP_UAVCAN* ap_uavcan, u
             }
         }
     }
+
+    struct DetectedModules tempslot;
+    //Sort based on the node_id, larger values first
+    //we do this, so that we have repeatable compass
+    //registration, especially in cases of extraneous
+    //CAN compass is connected.
```

Comment:
A device with lower node id can potentially show up before the one with higher node id. It can potentially replace the already set modules which is yet to be registered, hence leading to inconsistent behaviour.